src/components/countriesList/CountriesList.js

```diff
@@ -0,0 +1,35 @@
+import React, { Component } from 'react'
+
+import countries from '../../countries.json'
+
+import './CountriesList.css'
+
+import { NavLink } from 'react-router-dom'
+
+class CountriesList extends Component {
+
+    constructor() {
+        super()
+        this.state = {
+            countries: countries
+        }
+    }
+
+
+    render() {
+
+        return (
+
+            <>
+                <ul className="countriesUl">
+                    
+                    {this.state.countries.map(elm => <li><NavLink to={elm.cca3}  key={elm.cca3} >{elm.flag} {elm.name.official}</NavLink></li>)}
```

Comment:
```suggestion
                    {this.state.countries.map((elm, idx) => <li><NavLink key={idx} to={elm.cca3}  cca={elm.cca3} >{elm.flag} {elm.name.official}</NavLink></li>)}
```
Las keys ayudan a React a identificar que ítems han cambiado, son agregados, o son eliminados. Las keys deben ser dadas a los elementos dentro del array para darle a los elementos una identidad estable.

Cuando no tengas IDs estables para renderizar, puedes usar el índice del ítem como una key como último recurso.

No es recomendable usar índices para keys si el orden de los ítems puede cambiar. Esto puede impactar negativamente el rendimiento y puede causar problemas con el estado del componente.